koans/src/java7/AboutTryWithResources.java

```diff
@@ -0,0 +1,106 @@
+package java7;
+
+import com.sandwich.koan.Koan;
+
+import java.io.*;
+
+import static com.sandwich.koan.constant.KoanConstants.__;
+import static com.sandwich.util.Assert.assertEquals;
+
+public class AboutTryWithResources {
+
+    @Koan
+    public void lookMaNoClose() {
+        String str = "first line"
+                + System.lineSeparator()
+                + "second line";
+        InputStream is = new ByteArrayInputStream(str.getBytes());
+        String line;
+        /* BufferedReader implementing @see java.lang.AutoCloseable interface */
+        try (BufferedReader br =
+                     new BufferedReader(
+                             new InputStreamReader(is))) {
+            line = br.readLine();
+            //br guaranteed to be closed
```

Comment:
Hmmm... Great question. Since these koans are practically unit tests, it'd make sense to demonstrate behavior as opposed to merely commenting it. Please submit as an issue and I'll commit a change that asserts the stream is closed per a method that tries to read/write from/to a stream and deduces from the presence of an exception whether its closed (http://stackoverflow.com/questions/8655901/how-to-check-whether-an-outputstream-is-closed). Unless its fixed in the language the options are pretty much what you suggested with subclasses or a utility method. Reflection may be viable but that's prone to incompatibilities between the vendors JDK's.